### ISSUE TITLE
Rule suggestion: `block-no-unnecessary-child-block`

### ISSUE THREAD

**Issue by FloEdelmann** [stylelint-scss/stylelint-scss#679]
I suggest a new Stylelint rule that reports a single block inside another block:

```scss
// BAD
.foo {
    .bar { … }
}

// GOOD
.foo .bar { … }
```

```scss
// BAD
.foo {
    &-bar { … }
}

// GOOD
.foo-bar { … }
```

```scss
// GOOD, there are multiple child blocks
.foo {
    .bar { … }
    &-baz { … }
}
```

```scss
// GOOD, the outer selector contains a comma
.foo,
.bar {
    &-baz { … }
}
```

**Comment by kristerkari** [comment]
Sounds good!
